### PR TITLE
[bugfix] Make `attrs()` utility function pass when an `AttributeError` is raised

### PR DESCRIPTION
**Pull request by jjotero**
When the `LoggerAdapter` class calls the `attrs` method from `reframe/utility/__init__.py` it retrieves the values of the data descriptors from an object.

With the recent addition of the `variable` built-in, variables can be set as `required` (undefined), which means that a call to `getattr` on these variables will raise an `AttributeError`. When any of these required variables is not set when `LoggerAdapter` class attempts to fetch all the descriptors from a test, this raises a very ugly `AttributeError`.

For this reason, this `attrs` function from utility must pass if its call to `getattr` raises an `AttributeError`.

The unit test has also been changed accordingly.

### CODE REVIEW

**Review by vkarak — CHANGES_REQUESTED** [review]

File: reframe/utility/__init__.py [hunk 1]
```diff
@@ -287,9 +287,12 @@ def attrs(obj):
 
     # Look for descriptors
     for cls in type(obj).mro():
-        for attr, value in cls.__dict__.items():
-            if inspect.isdatadescriptor(value):
-                ret[attr] = getattr(obj, attr)
+        for attr in cls.__dict__:
+            if inspect.isdatadescriptor(cls.__dict__[attr]):
```
[the remark this answers is not in the document]
Comment (jjotero) [reply reframe/utility/__init__.py:291]: Using `items()` here will try to access the value of the descriptor and raise the `AttributeError`.

File: reframe/utility/__init__.py [hunk 2]
```diff
@@ -287,9 +287,12 @@ def attrs(obj):
...
+                try:
+                    ret[attr] = getattr(obj, attr)
+                except AttributeError:
+                    pass
```
[the remark this answers is not in the document]
Comment (jjotero) [reply reframe/utility/__init__.py:298]: Comments pushed.

**Review by ekouts — APPROVED** [review]

**Review by vkarak — APPROVED** [review]
lgtm

**Comment by vkarak** [comment]
@jenkins-cscs retry dom

**Pull request merged**